File: objectModel/Java/objectmodel/src/test/java/com/microsoft/commondatamodel/objectmodel/cdm/LocalEntityDeclarationDefinitionTest.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.cdm;

import com.microsoft.commondatamodel.objectmodel.TestHelper;
import org.testng.Assert;
import org.testng.annotations.Test;

public class LocalEntityDeclarationDefinitionTest {
  /**
   * Tests if the copy function creates copies of the sub objects
   */
  @Test
  public void testLocalEntityDeclarationDefinitionCopy() throws InterruptedException {
    CdmCorpusDefinition corpus = TestHelper.getLocalCorpus("", "testLocalEntityDeclarationDefinitionCopy", null, null, null, true);
    CdmLocalEntityDeclarationDefinition manifest = new CdmLocalEntityDeclarationDefinition(corpus.getCtx(), "name");

    final String dataPartitionName = "dataPartitionName";
    final String dataPartitionPatternName = "dataPartitionPatternName";
    final String incrementalPartitionName = "incrementalPartitionName";
    final String incrementalPartitionPatternName = "incrementalPartitionPatternName";

    CdmDataPartitionDefinition dataPartition = manifest.getDataPartitions().add(dataPartitionName);
    CdmDataPartitionPatternDefinition dataPartitionPattern = manifest.getDataPartitionPatterns().add(dataPartitionPatternName);
    CdmDataPartitionDefinition incrementalPartition = manifest.getIncrementalPartitions().add(incrementalPartitionName);
    CdmDataPartitionPatternDefinition incrementalPartitionPattern = manifest.getIncrementalPartitionPatterns().add(incrementalPartitionPatternName);

    CdmLocalEntityDeclarationDefinition copy = (CdmLocalEntityDeclarationDefinition) manifest.copy();
    copy.getDataPartitions().get(0).setName("newDataPartitionName");
    copy.getDataPartitionPatterns().get(0).setName("newDataPartitionPatternName");
    copy.getIncrementalPartitions().get(0).setName("newIncrementalPartitionName");
    copy.getIncrementalPartitionPatterns().get(0).setName("newIncrementalPartitionPatternName");

    Assert.assertEquals(dataPartition.getName(), dataPartitionName);
    Assert.assertEquals(dataPartitionPattern.getName(), dataPartitionPatternName);
    Assert.assertEquals(incrementalPartition.getName(), incrementalPartitionName);
    Assert.assertEquals(incrementalPartitionPattern.getName(), incrementalPartitionPatternName);
  }
}

File: objectModel/TypeScript/__test__/Cdm/LocalEntityDeclarationDefinition.test.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import {
    CdmCorpusDefinition,
    CdmLocalEntityDeclarationDefinition,
    CdmDataPartitionDefinition,
    CdmDataPartitionPatternDefinition
} from '../../internal';
import { testHelper } from '../testHelper';

describe('Cdm/LocalEntityDeclarationDefinitionTests', () => {
    /**
     * Tests if the copy function creates copies of the sub objects
     */
    it('testLocalEntityDeclarationDefinitionCopy', () => {
        const corpus: CdmCorpusDefinition = testHelper.getLocalCorpus('', 'testLocalEntityDeclarationDefinitionCopy', undefined, undefined, undefined, true);
        const entity: CdmLocalEntityDeclarationDefinition = new CdmLocalEntityDeclarationDefinition(corpus.ctx, 'name');

        const dataPartitionName: string = 'dataPartitionName';
        const dataPartitionPatternName: string = 'dataPartitionPatternName';
        const incrementalPartitionName: string = 'incrementalPartitionName';
        const incrementalPartitionPatternName: string = "incrementalPartitionPatternName";

        const dataPartition: CdmDataPartitionDefinition = entity.dataPartitions.push(dataPartitionName);
        const dataPartitionPattern: CdmDataPartitionPatternDefinition = entity.dataPartitionPatterns.push(dataPartitionPatternName);
        const incrementalPartition: CdmDataPartitionDefinition = entity.incrementalPartitions.push(incrementalPartitionName);
        const incrementalPartitionPattern: CdmDataPartitionPatternDefinition = entity.incrementalPartitionPatterns.push(incrementalPartitionPatternName);

        const copy: CdmLocalEntityDeclarationDefinition = entity.copy() as CdmLocalEntityDeclarationDefinition;
        copy.dataPartitions.allItems[0].name = 'newDataPartitionName';
        copy.dataPartitionPatterns.allItems[0].name = 'newDataPartitionPatternName';
        copy.incrementalPartitions.allItems[0].name = 'newIncrementalPartitionName';
        copy.incrementalPartitionPatterns.allItems[0].name = 'newIncrementalPartitionPatternName';

        expect(dataPartition.name)
            .toBe(dataPartitionName);
        expect(dataPartitionPattern.name)
            .toBe(dataPartitionPatternName);
        expect(incrementalPartition.name)
            .toBe(incrementalPartitionName);
        expect(incrementalPartitionPattern.name)
            .toBe(incrementalPartitionPatternName);
    });
});
